### ISSUE TITLE
Feature request: (repeating) byte offset header

### ISSUE THREAD

**Issue by c0d3z3r0** [8051Enthusiast/biodiff#19]
Very often I find myself counting (horizontal) byte offsets. While the hex space option already helps a lot, it's still not optimal. I would love to see a header like in quickly hacked-together mockup:

![image](https://github.com/8051Enthusiast/biodiff/assets/6205797/e30979ce-5d98-4b11-8f0d-78b85c22c8f7)

A second option could enable repeating of that header at a defined offset, e.g. each 0x100 bytes.

![image](https://github.com/8051Enthusiast/biodiff/assets/6205797/26cbe840-16d0-4765-a646-9516c9f5e2e7)

The code would need to respect options like hex spacer, column count (probably the most difficult one), vertical split. Also, I'm not sure what the best behaviour would be when scrolling up/down.

**Comment by 8051Enthusiast** [comment]
Sounds good! I was wondering how I'd deal with column counts that are not divisible by 16, but it seems that other hex editors like radare2 always just begin at `00` and go up from that instead of actually containing the last digits, which is the only sensible behavior in that case I guess.

> Also, I'm not sure what the best behaviour would be when scrolling up/down.

Do you mean whether the secondary headers should scroll with the content or stay at the fixed position on the screen?

**Comment by c0d3z3r0** [comment]
> > Also, I'm not sure what the best behaviour would be when scrolling up/down.
> 
> Do you mean whether the secondary headers should scroll with the content or stay at the fixed position on the screen?

Yep. Keeping it at a fixed position would require updating it (00 -> 10, 01 -> 11) or simply using x0, x1, x2 instead of 00, 01, .... Probably it's easier to let it move with the content?
